File: benchmarks/tsb/bench_styler.ts
/**
 * Benchmark: Styler — highlight max/min and background gradient on a 1000-row DataFrame
 */
import { DataFrame, dataFrameStyle } from "../../src/index.js";

const N = 1_000;
const WARMUP = 2;
const ITERATIONS = 5;

const a = Array.from({ length: N }, (_, i) => i * 1.0);
const b = Array.from({ length: N }, (_, i) => (N - i) * 2.0);
const c = Array.from({ length: N }, (_, i) => Math.sin(i / 100) * 100);
const df = DataFrame.fromColumns({ a, b, c });

for (let i = 0; i < WARMUP; i++) {
  dataFrameStyle(df).highlightMax().highlightMin().backgroundGradient().exportStyles();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  dataFrameStyle(df).highlightMax().highlightMin().backgroundGradient().exportStyles();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "styler",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_styler.py
"""Benchmark: Styler — highlight max/min and background gradient on a 1000-row DataFrame"""
import json
import time
import math
import pandas as pd
import numpy as np

N = 1_000
WARMUP = 2
ITERATIONS = 5

a = [i * 1.0 for i in range(N)]
b = [(N - i) * 2.0 for i in range(N)]
c = [math.sin(i / 100) * 100 for i in range(N)]
df = pd.DataFrame({"a": a, "b": b, "c": c})

def run_styler():
    styler = df.style.highlight_max().highlight_min().background_gradient()
    styler.to_html()  # force rendering

for _ in range(WARMUP):
    run_styler()

start = time.perf_counter()
for _ in range(ITERATIONS):
    run_styler()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "styler",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
